Keywords/webUtils/WebTableDataComparison.groovy

```diff
@@ -0,0 +1,209 @@
+	/**
+	 * Report the WebTable Map records differences
+	 * @param Key Name of WebTable record
+	 * @param Key Value of WebTable record
+	 * @param Recordentries differing Map object
+	 * @param Recordentries differing Right WebTable Map object
+	 * @param Recordentries differing of Left WebTable Map object
+	 */
+	public static void reportMapDifference(String recordKeyName, String recordKeyValue,
+			Map<String, MapDifference.ValueDifference<String>> entriesDiffering, Map<String, String> entriesOnlyOnLeft,
+			Map<String, String> entriesOnlyOnRight) {
+		try {
+			String failureMsg = "Value mismatch for field :: ";
+			for (String keyName : entriesOnlyOnLeft.keySet()) {
+				String actValue = entriesOnlyOnLeft.get(keyName);
+				String expValue = " ** NOT FOUND ** ";
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+			for (String keyName : entriesOnlyOnRight.keySet()) {
+				String actValue = " ** NOT FOUND ** ";
+				String expValue = entriesOnlyOnRight.get(keyName);
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+			for (String keyName : entriesDiffering.keySet()) {
+				String actValue = entriesDiffering.get(keyName).leftValue();
+				String expValue = entriesDiffering.get(keyName).rightValue();
+				KeywordUtil.markFailed("Failed " + recordKeyName + " :: " + recordKeyValue + " " + actValue + " " + expValue)
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			KeywordUtil.markFailed("Exception in reportMapDifference, unable to find differences in WebTables")
+		}
+	}
+
+	/**
+	 * Get the WebTable as Map
+	 * @param WebElement of WebTable
+	 * @param PrimaryKey of WebTable
+	 */
+	public static Map<String, Map<String, String>> getWebTableAsMap(WebElement table, String keyName) {
+		Map<String, Map<String, String>> webTableRecords = new LinkedHashMap();
+		int recordCounter = 0;
+		try {
+			//Get the WebTable into a List of String
+			List<String> webTableData = readWebTableData(table);
+			String colNames = webTableData.get(0);
+			String[] arrColName = colNames.split(",", -1);
+			int columnCount = arrColName.length;
+			for (String colValues : webTableData) {
+				String[] arrColValue = colValues.split(",", -1);
+				Map<String, String> fileRecord = new LinkedHashMap<>();
+				String mapKey = "";
+				for (int i = 0; i < columnCount; i++) {
+					String columnName = arrColName[i];
+					if (columnName.equalsIgnoreCase(keyName))
+						mapKey = arrColValue[i];
+					fileRecord.put(columnName, arrColValue[i]);
+				}
+				if (keyName.equalsIgnoreCase("DEFAULT") || mapKey.equalsIgnoreCase("")) {
+					webTableRecords.put(String.valueOf(recordCounter), fileRecord);
+					recordCounter++;
+				} else
+					webTableRecords.put(mapKey, fileRecord);
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			KeywordUtil.markFailed("Exception in getWebTableAsMap, unable to get WebTable as Map")
```

Comment:
WebUI.comment("Exception :: " + e.getMessage());
KeywordUtil.markFailedAndStop("Exception in getWebTableAsMap, unable to get Table as Map");

Reply:
Updated Exception Handling